ethindex/logdecode.py

```diff
@@ -0,0 +1,88 @@
+"""ethereum log decoding
+
+This module provides some helper functions to do ethereum log decoding
+
+See https://codeburst.io/deep-dive-into-ethereum-logs-a8d2047c7371 for a nice
+explanation.
+"""
+
+import hexbytes
+import eth_abi
+import itertools
+import eth_utils
+
+
+def build_address_to_abi_dict(addresses_json, compiled_contracts):
+    """build a contract-address to abi mapping from addresses.json and the contracts.json
+
+    This function doesn't read the json files, but instead expects the decoded
+    json from those files.
+    """
+    res = {}
+
+    def add_abi(a, n):
+        res[eth_utils.to_checksum_address(a)] = compiled_contracts[n]["abi"]
+
+    for network in addresses_json["networks"]:
+        add_abi(network, "CurrencyNetwork")
+
+    add_abi(addresses_json["unwEth"], "UnwEth")
+    add_abi(addresses_json["exchange"], "Exchange")
+    return res
+
+
+def decode_non_indexed_inputs(abi, log):
+    """decode non-indexed inputs from a log entry
+
+    This one decodes the values stored in the 'data' field."""
+    inputs = [x for x in abi["inputs"] if not x["indexed"]]
+    types = [x["type"] for x in inputs]
+    names = [x["name"] for x in inputs]
+    data = hexbytes.HexBytes(log["data"])
+    values = eth_abi.decode_abi(types, data)
+    return zip(names, values)
+
+
+def decode_indexed_inputs(abi, log):
+    """decode indexed inputs from a log entry
+
+    This one decodes the values stored in in topics fields (without topic 0)"""
+    inputs = [x for x in abi["inputs"] if x["indexed"]]
+    types = [x["type"] for x in inputs]
+    names = [x["name"] for x in inputs]
+    values = [eth_abi.decode_single(t, v) for t, v in zip(types, log["topics"][1:])]
+
+    return zip(names, values)
+
+
+def filter_events(abi):
```

Comment:
better function name. From filter_event I would expect a function that filters events, but this filters the abi for eventabis.
So maybe `get_eventabis_from_abi`? Or `get_event_abis`